java/src/main/java/com/genexus/webpanels/gridstate/GXGridStateHandler.java

```diff
@@ -114,10 +134,15 @@ public void clearFilterValues() {
 	}
 
 	public void addFilterValue(String name, String value) {
-		SdtGridState_InputValuesItem item = new SdtGridState_InputValuesItem(context);
-		item.setgxTv_SdtGridState_InputValuesItem_Value(value);
-		item.setgxTv_SdtGridState_InputValuesItem_Name(name);
-		state.getgxTv_SdtGridState_Inputvalues().add(item);
+		int idx = containsName(name);
+		if (idx > 0)
+			state.getgxTv_SdtGridState_Inputvalues().get(idx-1).setgxTv_SdtGridState_InputValuesItem_Value(value);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;value&#39; (index: 1 | type: String). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other validated parameters: &#39;name&#39;. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness (for example, using the @NonNull annotation as [described in the Lombok library](https://projectlombok.org/features/NonNull)), emptiness, or equality. Or to prevent direct calls to it, reduce the method&#39;s visibility.

Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).